### DISCUSSION TITLE
OKD 4.12 [UPI] Hang or freeze on bare metal machine when bond (802.3ad) NIC configuration

### DISCUSSION THREAD

**Discussion by roisdeveloper** [okd-project/okd#1583] [General]
**Describe the bug**

When adding new worker node (bare metal) with serveral NICs, the machine always stuck in **_'nodeip-configuration.service'_** when machine-os booting (2nd boot).  I used Fedora CoreOS fedora-coreos-37.20221127.3.0-live.x86_64 (live ISO) when installing to disk (success installed) and the NIC name detected (for bonding) are enp75s0f0 and enp75s0f1. 

When 2nd boot (applying machine-os), the machine stuck in nodeip-configuration.service. See picture below :
![image](https://user-images.githubusercontent.com/49701611/235277454-3ddb46f1-9d63-4994-86c2-2837298f7d65.png)

I tried to install another OKD 4.12 release version and got same problem. The latest version of machine-os (I've tried) was 37.20230322.3 with kernel 6.1.

I tried to run (live iso) fedora-coreos-37.20230322.3.0-live.x86_64, the machine was freeze after I config bonding interface (using nmcli). The machine became freeze after I run :

$ sudo nmcli conn modify bond0 connection.autoconnection-slave 1
$ sudo nmcli conn up bond0

I can't connect to network gateway.

I tried to install another OS, AlmaLinux 9.1 with kernel 5 upgraded to kernel 6.2, and try fedora-coreos-37.20230401.3.0-live.x86_64 (kernel 6.2), the NIC name changed.
**Before** kernel 6.2:
NIC 1 name : enp75s0f0
NIC 2 name : enp75s0f1

**In kernel 6.2** onward:
NIC 1 name : enp75s0f0np0
NIC 2 name : enp75s0f1np1

and success create bonding and connected to network gateway.

Do I have to wait for machine-os with kernel version 6.2 onward ?
Is it recommended if we make our own config machine-os with custom kernel ?

Notes:
- Worker node NIC type is Intel E810-XXVAM2.
- Master node installed successfully on another bare metal with another type (spec and NIC vendor).
- Adding new worker node using KVM with single NIC always success.

**Version**

4.12 (all 4.12.0-0.okd-2023 release versions) UPI install method on bare metal.

**How reproducible**

100%

**Log bundle**

_None_.

**Comment by vrutkovs** [comment]
So, kernel update has changed NIC naming?

>I used Fedora CoreOS fedora-coreos-37.20221127.3.0-live.x86_64 (live ISO) 

Check if using [latest FCOS stable](https://fedoraproject.org/coreos/download/?stream=stable) with kernel 6.2 avoids that?

**Reply by roisdeveloper** [reply]
To be more precise is starting from kernel 6.2, the NIC name changed on my bare metal machine. If I use the last version of fedora coreos with default kernel 6.2, I can configure network bonds without any problems. So I assume that os-machine (OKD 4.12 next release) will use kernel 6.2 and be able to solve this problem.

I've tried installing AlmaLinux 9.1 (on that machine) with the default kernel being 5.14, and the NIC name:
enp75s0f0 & enp75s0f0. It is the same if I use fedora-coreos-37.20221127.3.0-live.x86_64.

Then I upgraded to kernel 6.2 and kernel 6.3, and the NIC name was detected to be:
 enp75s0f0np0 & enp75s0f1np1.

So I decided to try running fedora-coreos-37.20230401.3.0-live.x86_64 with the default kernel 6.2, and the NIC name was immediately detected:
enp75s0f0np0 & enp75s0f1np1 and all network configurations run smoothly.

In summary :

fedora-coreos-37.20221127.3.0-live.x86_64 (OKD 4.12 based image) :

- NIC name detected enp75s0f0 & enp75s0f1
- All network configurations run smoothly
- Installing OS using --copy-network command.

fedora-coreos-37.20230303.3.0-live.x86_64 (live iso) :
- NIC name detected enp75s0f0 & enp75s0f1
- Bonding network configuration failed and made my machine stop working. 

fedora-coreos-37.20230322.3.0-live.x86_64 (live iso) :
- NIC name detected enp75s0f0 & enp75s0f1
- Bonding network configuration failed and made my machine stop working.

fedora-coreos-37.20230401.3.0-live.x86_64 (live iso & on disk) :
- NIC name detected enp75s0f0np0 & enp75s0f1np1
- Bonding network configuration run smoothly.

I'm also worried that even though machine-os has been updated, it still uses fedora-coreos-37.20221127.3.0-live.x86_64 as the based image, I still can't add workers using that machine, because the detected NIC name is different between fcos based image and machine-os.

Can we install OKD using a based image other than what this command suggests?
./openshift-install coreos print-stream-json | grep '\.iso[^.]'

**Reply by vrutkovs** [reply]
>Can we install OKD using a based image other than what this command suggests?

Yes, sure. Its merely a recommendation and default settings that we use in CI.

I'll update https://github.com/openshift/installer/pull/6902 to use newest FCOS stable to make kernel version is synchronised
